internal/pkg/scaffold/olm-catalog/csv_updaters.go

```diff
@@ -125,27 +129,20 @@ func (store *updaterStore) AddClusterRole(yamlDoc []byte) error {
 	return nil
 }
 
+const olmTNMeta = "metadata.annotations['olm.targetNamespaces']"
+
 func (store *updaterStore) AddDeploymentSpec(yamlDoc []byte) error {
 	dep := &appsv1.Deployment{}
 	if err := yaml.Unmarshal(yamlDoc, dep); err != nil {
 		return err
 	}
 
-	// OLM places the set of target namespaces for the operator in
-	// "metadata.annotations.olm.targetNamespaces". This value should be
-	// referred to in the Deployment spec's WATCH_NAMESPACE env.
-	for _, c := range dep.Spec.Template.Spec.Containers {
-		for _, e := range c.Env {
-			if e.Name == "WATCH_NAMESPACE" {
-				if e.ValueFrom == nil {
-					e.ValueFrom = &corev1.EnvVarSource{}
-				}
-				e.ValueFrom.FieldRef = &corev1.ObjectFieldSelector{
-					FieldPath: `metadata.annotations['olm.targetNamespaces']`,
-				}
-				break
-			}
-		}
+	// Make sure "olm.targetNamespaces" is referenced somewhere in dep,
+	// and emit a warning of not.
+	if !setAndCheckOLMNamespaces(dep) {
```

Comment:
Wdyt over has isValid and set funcs. Usually is not a good practice has func as deleteAndUpdate .. I mean without cohesion.

Reply:
I will refactor `WATCH_NAMESPACE` setting and the `olm.targetNamespaces` check into separate functions.